test/integration/pose_1_9_sdf.cc

```diff
@@ -0,0 +1,677 @@
+  poseElem->AddAttribute("rotation_type", "string", "rpy_radians", false);
+
+  sdf::ParamPtr poseValueParam = poseElem->GetValue();
+  ASSERT_NE(nullptr, poseValueParam);
+
+  ASSERT_TRUE(poseValueParam->Set<ignition::math::Pose3d>(
+        Pose(1, 2, 3, 0.4, 0.5, 0.6)));
+
+  Pose paramVal;
+  ASSERT_TRUE(poseValueParam->Get<ignition::math::Pose3d>(paramVal));
+  EXPECT_EQ(Pose(1, 2, 3, 0.4, 0.5, 0.6), paramVal);
+}
+
+//////////////////////////////////////////////////
+TEST(Pose1_9, PoseParamSetFromStringAndGet)
+{
+  using Pose = ignition::math::Pose3d;
+
+  sdf::ElementPtr poseElem(new sdf::Element);
+  poseElem->SetName("pose");
+  poseElem->AddValue("pose", "0 0 0   0 0 0", true);
+  poseElem->AddAttribute("relative_to", "string", "", false);
+  poseElem->AddAttribute("rotation_type", "string", "rpy_radians", false);
+
+  sdf::ParamPtr poseValueParam = poseElem->GetValue();
+  ASSERT_NE(nullptr, poseValueParam);
+
+  ASSERT_TRUE(poseValueParam->SetFromString("1 2 3   0.4 0.5 0.6"));
+
+  Pose paramVal;
+  ASSERT_TRUE(poseValueParam->Get<ignition::math::Pose3d>(paramVal));
+  EXPECT_EQ(Pose(1, 2, 3, 0.4, 0.5, 0.6), paramVal);
+}
+
+//////////////////////////////////////////////////
+TEST(Pose1_9, PoseParamSetAndElemGet)
+{
+  using Pose = ignition::math::Pose3d;
+
+  sdf::ElementPtr poseElem(new sdf::Element);
+  poseElem->SetName("pose");
+  poseElem->AddValue("pose", "0 0 0   0 0 0", true);
+  poseElem->AddAttribute("relative_to", "string", "", false);
+  poseElem->AddAttribute("rotation_type", "string", "rpy_radians", false);
+
+  sdf::ParamPtr poseValueParam = poseElem->GetValue();
+  ASSERT_NE(nullptr, poseValueParam);
+
+  ASSERT_TRUE(poseValueParam->Set<ignition::math::Pose3d>(
+        Pose(1, 2, 3, 0.4, 0.5, 0.6)));
+
+  Pose elemVal;
+  ASSERT_TRUE(poseElem->Get<ignition::math::Pose3d>("", elemVal, Pose()));
+  EXPECT_EQ(Pose(1, 2, 3, 0.4, 0.5, 0.6), elemVal);
+}
+
+//////////////////////////////////////////////////
+TEST(Pose1_9, PoseParamSetAndParentElemGet)
+{
+  using Pose = ignition::math::Pose3d;
+
+  sdf::ElementPtr poseElem(new sdf::Element);
+  poseElem->SetName("pose");
+  poseElem->AddValue("pose", "0 0 0   0 0 0", true);
+  poseElem->AddAttribute("relative_to", "string", "", false);
+  poseElem->AddAttribute("rotation_type", "string", "rpy_radians", false);
+
+  sdf::ParamPtr poseValueParam = poseElem->GetValue();
+  ASSERT_NE(nullptr, poseValueParam);
+
+  ASSERT_TRUE(poseValueParam->Set<ignition::math::Pose3d>(
+        Pose(1, 2, 3, 0.4, 0.5, 0.6)));
+
+  sdf::ElementPtr parentElem = poseValueParam->GetParentElement();
+  ASSERT_NE(nullptr, parentElem);
+  Pose parentElemVal;
+  ASSERT_TRUE(parentElem->Get<ignition::math::Pose3d>(
+      "", parentElemVal, Pose()));
+  EXPECT_EQ(Pose(1, 2, 3, 0.4, 0.5, 0.6), parentElemVal);
+}
+
+//////////////////////////////////////////////////
+TEST(Pose1_9, ChangingParentPoseElement)
+{
+  const double pi = 3.14159265358979323846;
```

Comment:
Hmm, since I was already converting degrees to radians myself without using `IGN_DTOR`, I thought it more appropriate to use raw values for pi too. Do you think I should still use `IGN_PI`?

Reply:
I guess that's a judgment call. I'd use `IGN_PI` since it's a well known constant whereas the other is a function. But'll I'll leave it up to you.